sdks/java/core/src/test/java/org/apache/beam/sdk/testing/PAssertTest.java

```diff
@@ -386,7 +386,7 @@ private static Throwable runExpectingAssertionFailure(Pipeline pipeline) {
     // is first caught by JUnit and causes a test failure.
     try {
       pipeline.run();
-    } catch (AssertionError exc) {
+    } catch (Exception exc) {
```

Comment:
For now, catching a `Throwable` and checking the error message is OK. But I wonder if it might be easy to polish this. In fact, it is [`Pipeline.run()` that unwraps `UserCodeException`](https://github.com/apache/incubator-beam/blob/f62d04e22679ee2ac19e3ae37dec487d953d51c1/sdks/java/core/src/main/java/org/apache/beam/sdk/Pipeline.java#L183) and `PAssert` failures should be an `AssertionError` wrapped in a `UserCodeException` which the runner should let through.

I'm a bit confused about the full state of things here, but it might actually be worth a little thread on the dev list.